test/test_tensorclass.py

```diff
@@ -67,6 +68,14 @@ def test_type():
     assert type(data) is MyDataUndecorated
 
 
+def test_signature():
+    sig = inspect.signature(MyData)
+    assert list(sig.parameters) == ["X", "y", "batch_size", "device", "_tensordict"]
+
+    with pytest.raises(TypeError, match="missing 2 required positional arguments"):
```

Comment:
I would probably add also other cases:
with just y
with tensordict
and with/without batch_size

Reply:
Good idea, added!

Reply:
you missed the x+y but without one of batch_size and device that should not raise in all cases

Reply:
I figured that those working cases should be adequately covered (implicitly at least) by all the other test cases. Can add an explicit one if you think it's required.

Reply:
I would say "better explicit than implicit" and I would add them. But I agree that from the coverage point of view they should be indirectly tested in other tests